lib/internal/errors.js

```diff
@@ -47,42 +47,112 @@ function lazyInternalUtil() {
   return internalUtil;
 }
 
-function makeNodeError(Base) {
-  return class NodeError extends Base {
-    constructor(key, ...args) {
-      super(message(key, args));
-      defineProperty(this, kCode, {
-        configurable: true,
-        enumerable: false,
-        value: key,
-        writable: true
-      });
-    }
+function sysErrorMessage(prefix, ctx) {
+  let message = `${prefix}: ${ctx.syscall} returns ` +
```

Comment:
not sure if it should be "returns" or "returned"

Reply:
@targos `returned` seems more appropriate, I'll change that, thanks!